autoload/rainbow_csv.vim

```diff
@@ -820,12 +818,13 @@ func! rainbow_csv#align_field(field, is_first_line, max_field_components_lens, i
     return repeat(' ', integer_delta_length) . clean_field . trailing_spaces
 endfunc
 
-
-func! rainbow_csv#csv_align()
+func! rainbow_csv#csv_align() range
+    let l:firstline = a:firstline
+    let l:lastline = a:lastline
     " The first (statistic) pass of the function takes about 40% of runtime, the second (actual align) pass around 60% of runtime.
     " Numeric-aware logic by itself adds about 50% runtime compared to the basic string-based field width alignment
     " If there are lot of numeric columns this can additionally increase runtime by another 50% or more.
-    let show_progress_bar = wordcount()['bytes'] > 200000
+    let show_progress_bar = (l:lastline - l:firstline) > 200000
     let [delim, policy, comment_prefix] = rainbow_csv#get_current_dialect()
     if policy == 'monocolumn'
         echoerr "RainbowAlign is available only for highlighted CSV files"
```

Comment:
Instead of immediately returning you can adjust delim and policy to '|' and simple for markdown files - this would be the special case at Align level only. same for the shrink. You can also check here that range is provided instead of checking it below.

Reply:
done